src/main.rs

```diff
@@ -68,28 +69,48 @@ fn run_command(command: SubCommand, now: DateTime<Local>) {
         let day: Day = possible_day.unwrap();
 
         match command {
-            SubCommand::Out => punch_out(&now, day),
-            SubCommand::Pause=> take_break(&now, day),
-            SubCommand::Resume => resume(&now, day),
-            SubCommand::Summary => summary(&now, day),
-            SubCommand::View => view_day(day),
-            SubCommand::Edit => edit_day(day),
-            SubCommand::In | SubCommand::Invalid(_) => unreachable!("We shouldn't be processing punch in here"),
+            SubCommand::Out(_) => punch_out(&now, day),
+            SubCommand::Pause(_) => take_break(&now, day),
+            SubCommand::Resume(_) => resume(&now, day),
+            SubCommand::Summary(_) => summary(&now, day),
+            SubCommand::View(_) => view_day(day),
+            SubCommand::Edit(_) => edit_day(day),
+            SubCommand::In(_) | SubCommand::Invalid(_) => unreachable!("We shouldn't be processing punch in here"),
         }
     }
 }
 
-fn punch_in(now: &DateTime<Local>) {
+fn punch_in(now: &DateTime<Local>, other_args: Vec<String>) {
     if let Ok(_) = read_day(now) {
         println!("You've already clocked in for the day!");
     }
-    else {
-        let new_day: Day = Day::new(&now);
+    else{
+        let time_to_do: u64 = get_time_to_do_for_day(other_args);
+        let new_day: Day = Day::new(&now, time_to_do);
         println!("Clocking in for the day at '{}'", &new_day.get_day_start_as_str());
         write_day(&new_day);
     }
 }
 
+fn get_time_to_do_for_day(other_args: Vec<String>) -> u64 {
+    if other_args.len() == 0 {
+        return get_default_day_in_minutes();
+    }
+    let first_arg: Result<u64, std::num::ParseIntError> = other_args[0].parse::<u64>();
+    return match first_arg {
+        Ok(ttd) => ttd,
+        Err(_) => {
+            let ttd: u64 = get_default_day_in_minutes();
+            println!("'{}' is not a valid value for time to do today. Using default instead ({})", other_args[0], ttd);
```

Comment:
It's worth mentioning to the user that the time to do can be edited using `punch edit`